eZ/Publish/Core/MVC/Symfony/SiteAccess/Matcher/HostElement.php

```diff
@@ -37,10 +37,15 @@ class HostElement implements VersatileMatcher
     /**
      * Constructor.
      *
-     * @param int $elementNumber Number of elements to take into account.
+     * @param int|array $elementNumber Number of elements to take into account.
      */
     public function __construct($elementNumber)
     {
+        if (is_array($elementNumber) && isset($elementNumber['value'])) {
```

Comment:
I prefer option 2. But it can not be backported.

So if possible I would have option 1 for 5.4 and 1.0.x (maybe with depreciation notice) and option 2 for the next release (with bc note).

ping @bdunogier